gulpfile.js

```diff
@@ -286,22 +308,37 @@ gulp.task('prefetch', function(cb) {
           // download only what can be included in offline installer
           if (reqs[currentKey].bundle === 'yes') {
             counter++;
-            console.log('DOWNLOADING -> ' + currentUrl);
-            request(currentUrl)
-              .pipe(fs.createWriteStream(currentFile)).on('finish',function() {
-                // create a sha256sum file
-                counter--;
-                if(counter===0) {
-                  cb();
-                }
-              });
+            promises[counter] = new Promise(function(resolve, reject) {
+              console.log('DOWNLOADING -> ' + currentUrl);
+              request(currentUrl)
+                .pipe(fs.createWriteStream(currentFile)).on('finish',function(err, res) {
+                  if (err) { 
+                    reject(err);
+                  } else {
+                    // console.log('DOWNLOADED <- ' + currentFile);
+                    counter++;
+                    promises[counter] = copyFromPrefetchToDist(currentFile, destFile);
+                    counter--;
+                    resolve(res);
+                  }
+                });
+            });
           }
         } else {
-          console.log('DOWNLOADED <- ' + currentFile);
+          counter++;
+          promises[counter] = copyFromPrefetchToDist(currentFile, destFile);
+          counter--;
         }
       });
+
     }
   }
+  promise.all(promises);

```

Comment:
Nick it should be something like this snipped below and coounter would not be neded anymore 
``` JavaScript
promise.all(promises).then((result)=> {
  cb();
}).catch((error)=>{
  cb(error);
});
```